contracts/rollup/Rollup.sol

```diff
@@ -419,7 +422,10 @@ contract Rollup is BatchManager, EIP712, IEIP712 {
         if (
             result != Types.Result.Ok ||
             (processedStateRoot != target.commitment.stateRoot)
-        ) startRollingBack(batchID);
+        ) {
+            emit RollbackTriggered(batchID, result);
+            startRollingBack(batchID);
+        }
```

Comment:
right now i have added this `emit` statement wherever the `startRollingBack` function is called. Another way could be to emit this event inside the `startRollingBack` function itself, but that would require a change in the function signature to include the `result` as an argument and subsequent changes wherever this function is called. 
Is the current approach fine?

Reply:
Good question. I would vote to change the function signature of startRollingBack since the RollbackTriggered should be emitted wherever startRollingBack is called. It also makes sense for the startRollingBack to take the result as an argument, as `result` tells the `reason` why we start rolling back.